src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,176 @@
+package task;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import exception.JarvisException;
+
+public class TaskList {
+    private final List<Task> tasks;
+
+    /**
+     * Constructs a TaskList object.
+     *
+     * @param tasks The initial task list.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public List<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Returns the current task list as a formatted string.
+     *
+     * @return The formatted task list.
+     */
+    public String printTasks() {
+        int count = tasks.size();
+        if (count == 0) {
+            return "You have no tasks in your list. :-)";
+        }
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < count; i++) {
+            sb.append(i + 1).append(".\t").append(tasks.get(i));
+            if (i + 1 != count) {
+                sb.append("\n");
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param parsedCommand The input command containing the task number.
+     * @throws JarvisException If task number is invalid.
+     */
+    public String markAsDone(HashMap<String, Object> parsedCommand) throws JarvisException {
+        checkIfEmpty();
+        try {
```

Comment:
Would it be easier if the singular task and the list tasks have a more distinct name from each other? That way it can be identified with one glance.